### ISSUE TITLE
Allow variable predicates in queries

### ISSUE THREAD

**Issue by gtfierro** [gtfierro/hod#3]
Handling arbitrary chains of predicates is tricky, so focus on one predicate var per term for now (possibly including `+`,`?`,`*` support if it turns out to not be too tricky)

- [ ] `?s ?p ?o` (7461d2eba84954c58c1c2f3b1fc0fa0c41eca233, e6f234682420fd5bd1ef5a04cd62d04d736d6d2d, and ef66374c5a29ace6ea2d7667bde300be35106b63 so far)
- [x] `?s ?p o` (3acf8cd9bf83c11f8dbb1ba51721b6d4f1f71e92)
- [x] `s ?p ?o` (3acf8cd9bf83c11f8dbb1ba51721b6d4f1f71e92)
- [x] `s ?p o` (007a64b495f651a3c3a3a2833a041a89e06d6162)

**Comment by gtfierro** [comment]
We can move ahead with implementation of #4, even though the `?s ?p ?o` pattern is still not implemented.

This last pattern is a little trickier because it involves changing the query planner to be aware of terms with 3 variables, which wasn't In the original design.

**Comment by gtfierro** [comment]
Turns out we did need the `?s ?p ?o` pattern, but it was simple to do provided we had constrained at least one of the variables. The last part to do is the simple dump of the triples in the graph, which requires writing some new traversal operators. These should be easy because its just iterating over the underlying LevelDB, but we need to give some thought on how to iterate:
- do we list the triples as they were inserted?
- do we only want to list triples that are in the building?
- do we want to ignore the 'inverse' triples?

We don't need this functionality just yet so I'm punting on implementing it.
